### PR TITLE
[TT-7418] Create OAS API version atomically in OSS

### PR DESCRIPTION
**Pull request by furkansenharputlu**
https://tyktech.atlassian.net/browse/TT-7418

### CODE REVIEW

File: gateway/api.go [hunk 1]
```diff
@@ -1123,7 +1123,11 @@ func (gw *Gateway) handleAddApi(r *http.Request, fs afero.Fs, oasEndpoint bool)
 				baseAPI.VersionDefinition.Default = newVersionName
 			}
 
-			if !oasEndpoint {
+			if baseAPI.IsOAS {
+				baseAPI.OAS.Fill(*baseAPI.APIDefinition)
+				err, _ := gw.writeOASAndAPIDefToFile(fs, baseAPI.APIDefinition, &baseAPI.OAS)
+				log.WithError(err).Errorf("Error occurred while updating base OAS API with id: %s", baseAPI.APIID)
```
Comment (titpetric) [inline gateway/api.go:1129]: This would log even if err=nil

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4581/merge
Commit: c02265ca74e4eda901150ca0c6c06e2e5db4f0b1 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3780074136)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4581/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3780065876)

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4581/merge
Commit: ee7c1a920784c54e1b6d68cbd75b899eb789f9ae 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3780156819)

**Review by Yasintrm — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 4cc3d4baea6252f8fad37206ffa2257282d2906f [TT-7418] Create OAS API version atomically in OSS (#4581)

https://tyktech.atlassian.net/browse/TT-7418
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3780302130)

**Pull request merged**